src/eventDispatchers/mouseEventHandlers/mouseDown.js

```diff
@@ -29,122 +30,193 @@ export default function (evt) {
 
   let tools;
   const eventData = evt.detail;
-  const element = eventData.element;
-  const coords = eventData.currentPoints.canvas;
-  const distance = 6;
+  const element = evt.detail.element;
+  const coords = evt.detail.currentPoints.canvas;
 
-  // TODO: instead of filtering these for every interaction, we can change our
-  // State's structure to always know these values.
-  // Filter out disabled and enabled
+  // High level filtering
   tools = getInteractiveToolsForElement(element, getters.mouseTools());
-  // Filter out tools that do not match mouseButtonMask
   tools = tools.filter((tool) =>
     isMouseButtonEnabled(eventData.which, tool.options.mouseButtonMask)
   );
-  tools = getToolsWithDataForElement(element, tools);
 
-  // Find tools with handles we can move
-  const toolsWithMoveableHandles = tools.filter((tool) => {
-    const toolState = getToolState(eventData.element, tool.name);
+  // Check if any tool has a special reason to grab the current event
+  const specialTools = tools.filter(
```

Comment:
I'm happy with the way this is done, and understand this is cleaning up what I did, however, upon further reflection, would it not be better to have either:
- "special tools" extend a subclass of specialTool, in which these methods are mandatory, so we don't need to constantly check if all these conditions are true.
- A simple tag, isSpecialTool (or some better name), that implies these methods are present.

I'd prefer the former as its more vigorous, but I'm not sure if it adds another layer of complexity for DX. I think we could convey it quite easily though with appropriate naming and warning functions.

Reply:
Edit: quick mockup if we went with my first solution:

```
const specialTools = tools.filter(
  (tool) => tool instanceof 'SpecialTool' &&
  tool.isValidTarget(evt)
);
```